canonicalwebteam/blog/django/views.py

```diff
@@ -86,6 +90,68 @@ def group(request, slug, template_path):
     return render(request, template_path, context)
 
 
+def archives(request, template_path="blog/archives.html"):
+    try:
+        page_param = request.GET.get("page", default="1")
+        category_param = request.GET.get("category", default="")
+        group_param = request.GET.get("group", default="")
+        month_param = request.GET.get("month", default="")
+        year_param = request.GET.get("year", default="")
+
+        group = None
+        group_id = ""
+        if group_param != "":
+            group = api.get_group_by_slug(group_param)
+            group_id = group["id"]
```

Comment:
I would declare `groups and categories = []` beforehand and then append the id to it inside the if.

Reply:
I have added sth like this (a memory based cache) in https://github.com/canonical-web-and-design/canonicalwebteam.blog/pull/45 . 

Mind giving that review? Then I can rebase this branch on top of it.